components/github/githubService.js

```diff
@@ -0,0 +1,120 @@
+import config from 'config-yml'
+import { driver } from '@rocket.chat/sdk'
+import { dal as interactionDAL } from '../interactions'
+
+const getId = data => {
+  return data.pull_request && data.pull_request.merged
+    ? data.pull_request.user.id
+    : data.sender.id
+}
+
+const getTypeByAction = action => {
+  let res = null
+  switch (action) {
+    case 'closed':
+      res = 'merged_pull_request'
+      break
+    case 'opened':
+      res = 'pull_request'
+      break
+  }
+  return res
+}
+
+const getType = data => {
+  let type = null
+  if (data.issue) type = 'issue'
+  if (data.review) type = 'review'
+  if (data.pull_request) type = defaultFunctions.getTypeByAction(data.action)
```

Comment:
é essa ordem mesmo? entao o pullrequest sempre tera preferencia podendo sobrescrever os outros casos?
pq caso for poderia ser algo
return (data.pull_request && defaultFunctions.getTypeByAction(data.action))
 || (data.review && 'review)
 || (data.issue && 'issue')

em um unico fluxo, sem leitura induzindo a outros fluxos, ficando mais legivel